bgpd/bgp_routemap.c

```diff
@@ -5130,6 +5183,38 @@ DEFUN_YANG (no_set_distance,
 	return nb_cli_apply_changes(vty, NULL);
 }
 
+DEFUN_YANG(set_l3vpn_nexthop_encapsulation, set_l3vpn_nexthop_encapsulation_cmd,
+	   "[no] set l3vpn next-hop encapsulation gre",
+	   NO_STR SET_STR
+	   "L3VPN operations\n"
+	   "Next hop Information\n"
+	   "Encapsulation options (for BGP only)\n"
+	   "Accept L3VPN traffic over GRE encapsulation\n")
+{
+	const char *xpath =
+		"./set-action[action='frr-bgp-route-map:set-l3vpn-nexthop-encapsulation']";
+	int xpath_len = sizeof(strlen(xpath));
+	char xpath_value[XPATH_MAXLEN + xpath_len];
```

Comment:
`XPATH_MAXLEN` is the *maximum length*, it is nonsense to allocate a buffer intended to store an Xpath whose size is larger than the maximum Xpath length. You need to use `XPATH_MAXLEN` as the buffer size, and later on check the return of `snprintf` and handle the case where the path is too long to fit in the buffer.